### DISCUSSION TITLE
An anisotropic mobility question

### DISCUSSION THREAD

**Discussion by dongcongcong606** [idaholab/moose#18346] [Q&A Modules: Phase field] [answered]
Dear Friends!
 Hello everyone! I am a beginner to learn MOOSE. I would like to ask you some questions, and I wish you can give me some guidance or advice, even though these questions seem simple to you. I have recently worked to develop a program on mobility anisotropy. I'm going to use the C-H equation. I introduced an anisotropic function about the angle in mobility and defined a corresponding subroutine in the ''Material' module. Next, I defined a subroutine in the kernel that introduced anisotropic functions modeled after "SplitCHWResBase". But I met some problems, as follows:
①Do I also need to copy "SplitCHParsed" to create a new subroutine?
②How to solve the "Signal 11" problem? (I have tried to check the problem according to Q&A, but failed)
③How do I know if a subroutine module I've written is successful?
（I enclose the associated subroutine defined below）
[ANisotropicFirst_C.txt](https://github.com/idaholab/moose/files/6822092/ANisotropicFirst_C.txt)
[ANisotropicFirst_h.txt](https://github.com/idaholab/moose/files/6822093/ANisotropicFirst_h.txt)
[SplitCHWResBase2.txt](https://github.com/idaholab/moose/files/6822119/SplitCHWResBase2.txt)
![20210715001](https://user-images.githubusercontent.com/82868802/125768503-c2012db1-c0f0-4a20-8802-eb3187e11664.png)

**Comment by WilkAndy** [comment, marked as answer]
I found these pages very helpful when starting to learn C++:  https://mooseframework.inl.gov/getting_started/examples_and_tutorials/index.html
